### PR TITLE
feat(TSCH-34): add all features of ActionBar to List.Toolbar

### PR DESCRIPTION
**Pull request by mtikhinskaya**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

As ActionBar is used in List.Toolbar and is not obligatory there I remove all required props from ActionBar. In future the Toolbar propTypes should be refactor. Then we return required props to ActionBar

### CODE REVIEW

File: src/ActionBar/ActionBar.component.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ const actionsShape = {
 
 ActionBar.propTypes = {
 	selected: PropTypes.number,
-	actions: PropTypes.shape(actionsShape).isRequired,
```
Comment (jmfrancois) [inline src/ActionBar/ActionBar.component.js]: if actions is not required, the actions.map will fails

https://github.com/Talend/react-talend-components/pull/167/files#diff-6fb482b40d35c67188421016e1af547dR10

I think this is a good idea so add the corresponding test case and fix the code to make it work with undefined

File: stories/List.js [hunk 2]
```diff
@@ -89,7 +89,7 @@ const props = {
 		},
 		itemProps: {
 			classNameKey: 'className',
-			isSelected: item => selected.find(next => next.id === item.id),
+			isSelected: item => false,
```
Comment (jmfrancois) [inline stories/List.js]: just remove the isSelected arrow function do the same

Comment (mtikhinskaya) [inline stories/List.js]: if I remove it the checkbox render incorrect

File: src/ActionBar/ActionBar.component.js (hunk 1 above)
Comment (mtikhinskaya) [inline src/ActionBar/ActionBar.component.js]: Done

File: src/ActionBar/ActionBar.component.js [hunk 3]
```diff
@@ -15,25 +15,27 @@ function ActionBar({ selected, actions, multiSelectActions }) {
...
+			}
+			if (actions.right) {
+				actionBar.push(
+					<div key={1} className={classNames(css['navbar-right'], 'navbar-right')}>
+						{ selected > 0 && !actions.left ? renderSelectedCount() : null}
+						{ renderActions(actions.right) }
+					</div>
+				);
+			}
+			return actionBar;
```
Comment (jsomsanith) [inline src/ActionBar/ActionBar.component.js]: There is a linter error : you need to return something even outside of the if `return null;`
Even better, a fast return
```
if (!actions) {
	return null;
}
... (rest of the render code)
```

File: src/List/Toolbar/Toolbar.component.js [hunk 4]
```diff
@@ -41,29 +41,38 @@ function hasProps(props) {
...
-	let actions = props.actions;
-	if (props.id && actions) {
-		actions = {
-			left: adaptActionsIds(actions.left, props.id),
-			right: adaptActionsIds(actions.right, props.id),
-		};
-	}
-
-	const hasDisplayMode = hasProps(displayProps);
+	const hasActionBarPros = hasProps(actionBarProps);
```
Comment (jsomsanith) [inline src/List/Toolbar/Toolbar.component.js]: A little typo : `Pro**p**s`

File: stories/List.js (hunk 2 above)
Comment (jsomsanith) [inline stories/List.js]: In this case, add a story with the selection feature that is disabled please

File: src/ActionBar/ActionBar.snapshot.test.js [hunk 5]
```diff
@@ -19,6 +19,13 @@ const secondaryAction = {
 };
 
 describe('ActionBar', () => {
+	it('should render without actions', () => {
+		//when
```
Comment (jsomsanith) [inline src/ActionBar/ActionBar.snapshot.test.js]: Linter error : add a space after `//`

File: src/ActionBar/ActionBar.snapshot.test.js [hunk 6]
```diff
@@ -19,6 +19,13 @@ const secondaryAction = {
...
+		const wrapper = renderer.create(<ActionBar />).toJSON();
+		//then
```
Comment (jsomsanith) [inline src/ActionBar/ActionBar.snapshot.test.js]: Linter error : add a space after `//`

File: src/ActionBar/ActionBar.component.js [hunk 7]
```diff
@@ -15,25 +15,27 @@ function ActionBar({ selected, actions, multiSelectActions }) {
...
-		const { left, right } = getActionsToRender();
-		const actionBar = [];
-		if (left) {
-			actionBar.push(
-				<div key={0} className={classNames('navbar-left')}>
-					{ selected > 0 ? renderSelectedCount() : null}
-					{ renderActions(left) }
-				</div>
-			);
+		const actions = getActionsToRender();
```
Comment (jsomsanith) [inline src/ActionBar/ActionBar.component.js]: Linter error : 'actions' is already declared in the upper scope
Just name variables differently than existing variable names.

**Pull request merged**